storage/grpc_metrics.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"strings"
+
+	mexporter "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/metric"
+	"github.com/google/uuid"
+	"go.opentelemetry.io/contrib/detectors/gcp"
+	"go.opentelemetry.io/otel/attribute"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"google.golang.org/api/option"
+	estats "google.golang.org/grpc/experimental/stats"
+	"google.golang.org/grpc/stats/opentelemetry"
+)
+
+const (
+	monitored_resource_name         = "storage.googleapis.com/Client"
+	mr_location_label_default       = "global"
+	mr_cloud_platform_label_default = "unknown"
+	mr_host_id_label_default        = "unknown"
+	metric_format_prefix            = "storage.googleapis.com/client/"
+	metric_lb_locality_label        = "grpc.lb.locality"
+)
+
+func getMetricsToEnable() []estats.Metric {
```

Comment:
This and `getMetricsEnabledByDefault` could probably be local vars rather than functions given that they are just static data.